exercises/doubly-linked-list/src/lib.rs

```diff
@@ -42,6 +42,7 @@ impl<T> Cursor<'_, T> {
         unimplemented!()
     }
 
+    #[allow(clippy::should_implement_trait)]
     /// Move one position forward (towards the back) and
     /// return a reference to the new position
     pub fn next(&mut self) -> Option<&mut T> {
```

Comment:
```suggestion
    /// Move one position forward (towards the back) and
    /// return a reference to the new position
    #[allow(clippy::should_implement_trait)]
    pub fn next(&mut self) -> Option<&mut T> {
```

Purely a style thing, but the convention I prefer is that attributes come after doc-comments. The reasoning is that doc comments can get very long, but there are very rarely more than a few lines of attributes. Putting the doc comment first ensures that attributes are visible when looking at the top of the definition.